crates/bevy_ui/src/flex/mod.rs

```diff
@@ -216,14 +217,18 @@ pub fn flex_node_system(
     // assume one window for time being...
     let logical_to_physical_factor = windows.scale_factor(WindowId::primary());
 
-    if scale_factor_events.iter().next_back().is_some() {
+    if scale_factor_events.iter().next_back().is_some() || ui_scale.is_changed() {
         update_changed(
             &mut *flex_surface,
-            logical_to_physical_factor,
+            logical_to_physical_factor * ui_scale.scale,
```

Comment:
Given that this calculation is done in a few places, i think it makes sense to factor this out into a `scale_factor` variable thats used everywhere.